Flight Entity Constantly Crashing the Server

### What happened?

[22:12:31] [Server thread/ERROR]: Exception stopping the server
net.minecraft.class_148: Saving entity NBT
        at net.minecraft.class_1297.method_5647(class_1297.java:1844) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5786(class_1297.java:1762) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5662(class_1297.java:1770) ~[server-intermediary.jar:?]
        at net.minecraft.class_5565.method_31734(class_5565.java:108) ~[server-intermediary.jar:?]
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708) ~[?:?]
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?]
        at net.minecraft.class_5565.method_31760(class_5565.java:106) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31812(class_5579.java:299) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31837(class_5579.java:316) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31842(class_5579.java:394) ~[server-intermediary.jar:?]
        at it.unimi.dsi.fastutil.longs.LongCollection.removeIf(LongCollection.java:274) ~[fastutil-8.5.9.jar:?]
        at net.minecraft.class_5579.method_31836(class_5579.java:392) ~[server-intermediary.jar:?]
        at net.minecraft.class_3218.method_14176(class_3218.java:792) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3723(MinecraftServer.java:536) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3782(MinecraftServer.java:602) ~[server-intermediary.jar:?]
        at net.minecraft.class_3176.method_3782(class_3176.java:537) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:703) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "dev.amble.ait.core.tardis.Tardis.getDesktop()" because the return value of "dev.amble.ait.api.tardis.link.v2.TardisRef.get()" is null
        at dev.amble.ait.core.entities.FlightTardisEntity.method_5652(FlightTardisEntity.java:287) ~[ait-1.2.0.2757-1.20.1-beta.jar:?]
        at net.minecraft.class_1297.method_5647(class_1297.java:1826) ~[server-intermediary.jar:?]
        ... 18 more

### Version

1.2.0

### Build number & branch

1.2.0

### Logs & screenshots

[22:12:31] [Server thread/ERROR]: Exception stopping the server
net.minecraft.class_148: Saving entity NBT
        at net.minecraft.class_1297.method_5647(class_1297.java:1844) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5786(class_1297.java:1762) ~[server-intermediary.jar:?]
        at net.minecraft.class_1297.method_5662(class_1297.java:1770) ~[server-intermediary.jar:?]
        at net.minecraft.class_5565.method_31734(class_5565.java:108) ~[server-intermediary.jar:?]
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708) ~[?:?]
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?]
        at net.minecraft.class_5565.method_31760(class_5565.java:106) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31812(class_5579.java:299) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31837(class_5579.java:316) ~[server-intermediary.jar:?]
        at net.minecraft.class_5579.method_31842(class_5579.java:394) ~[server-intermediary.jar:?]
        at it.unimi.dsi.fastutil.longs.LongCollection.removeIf(LongCollection.java:274) ~[fastutil-8.5.9.jar:?]
        at net.minecraft.class_5579.method_31836(class_5579.java:392) ~[server-intermediary.jar:?]
        at net.minecraft.class_3218.method_14176(class_3218.java:792) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3723(MinecraftServer.java:536) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_3782(MinecraftServer.java:602) ~[server-intermediary.jar:?]
        at net.minecraft.class_3176.method_3782(class_3176.java:537) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:703) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "dev.amble.ait.core.tardis.Tardis.getDesktop()" because the return value of "dev.amble.ait.api.tardis.link.v2.TardisRef.get()" is null
        at dev.amble.ait.core.entities.FlightTardisEntity.method_5652(FlightTardisEntity.java:287) ~[ait-1.2.0.2757-1.20.1-beta.jar:?]
        at net.minecraft.class_1297.method_5647(class_1297.java:1826) ~[server-intermediary.jar:?]
        ... 18 more

### I've checked the reported issues...

- [x] ...and this issue was not reported before.

--- a/src/main/java/dev/amble/ait/core/entities/FlightTardisEntity.java
+++ b/src/main/java/dev/amble/ait/core/entities/FlightTardisEntity.java
@@ -276,13 +276,19 @@ public class FlightTardisEntity extends LinkableLivingEntity implements JumpingM
     @Override
     public void readCustomDataFromNbt(NbtCompound nbt) {
         super.readCustomDataFromNbt(nbt);
-        this.interiorPos = BlockPos.fromLong(nbt.getLong("InteriorPos"));
+
+        if (nbt.contains("InteriorPos")) {
+            this.interiorPos = BlockPos.fromLong(nbt.getLong("InteriorPos"));
+        } else {
+            this.interiorPos = BlockPos.ORIGIN;
+        }
     }
 
     @Override
     public void writeCustomDataToNbt(NbtCompound nbt) {
         super.writeCustomDataToNbt(nbt);
         if (this.getWorld().isClient()) return;
+        if (!this.isLinked()) return;
 
         BlockPos consolePos = this.tardis().get().getDesktop().getConsolePos().iterator().next();
         BlockPos pos = WorldUtil.findSafeXZ(this.tardis().get().asServer().getInteriorWorld(), consolePos, 2);

